### PR TITLE
feat: associate a label to an api deployment

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes https://github.com/gravitee-io/issues/issues/4742

### CODE REVIEW

File: src/management/api/apiAdmin.controller.ts [hunk 1]
```diff
@@ -119,17 +119,18 @@ class ApiAdminController {
...
       clickOutsideToClose: true,
       locals: {
         title: 'Would you like to deploy your API?',
+        msg: 'You can provide a label to identify your deployment',
         confirmButton: 'OK'
       }
     }).then((response) => {
-      if (response) {
-        this.deploy(api);
+      if (response || response === '') {
```
[the remark this answers is not in the document]
Comment (ytvnr) [reply src/management/api/apiAdmin.controller.ts:132]: `if( '' )` evaluates to `false`. 

The dialog returns `false`only if we cancel the dialog. So, here I need to test if I have a value or if this value is empty before calling the backend.

File: src/management/management.module.ts [hunk 2]
```diff
@@ -87,6 +87,7 @@ import ApiCreationStep4Component from '../management/api/creation/steps/api-crea
 import ApiCreationStep5Component from '../management/api/creation/steps/api-creation-step5.component';
 import ApiImportComponent from '../components/import/import-api.component';
 import NewApiController from '../management/api/creation/newApiPortal.controller';
+import DialogConfirmDeploymentController from './api/deploy/confirmDeploymentDialog.controller';
```
[the remark this answers is not in the document]
Comment (ytvnr) [reply src/management/management.module.ts:90]: I understand this comment for consistency with previous lines, but why ? (we already are in 'management' folder

File: src/management/platform/home/home.html [hunk 3]
```diff
@@ -50,14 +50,22 @@ <h2>API Events</h2>
...
             </tr>
             </thead>
             <tbody md-body>
             <tr md-row ng-repeat="event in homeCtrl.events.content" style="height: 25px;">
+              <td md-cell>
+                <span class="badge" ng-if="event.properties.deployment_number"># {{event.properties.deployment_number}}</span>
+              </td>
               <td md-cell>{{event.updated_at | humanDatetimeFilter }}</td>
+              <td md-cell>
+                <span>{{event.properties.deployment_label}}</span>
```
[the remark this answers is not in the document]
Comment (ytvnr) [reply src/management/platform/home/home.html]: no ! resolved by 5272088

File: src/management/platform/dashboard/dashboard.html [hunk 4]
```diff
@@ -51,14 +51,22 @@ <h2>API Events</h2>
...
             </tr>
             </thead>
             <tbody md-body>
             <tr md-row ng-repeat="event in dashboardCtrl.events.content" style="height: 25px;">
+              <td md-cell>
+                <span class="badge" ng-if="event.properties.deployment_number"># {{event.properties.deployment_number}}</span>
+              </td>
               <td md-cell>{{event.updated_at | humanDatetimeFilter }}</td>
+              <td md-cell>
+                <span>{{event.properties.deployment_label}}</span>
```
[the remark this answers is not in the document]
Comment (ytvnr) [reply src/management/platform/dashboard/dashboard.html]: no ! resolved by 5272088

File: src/management/management.module.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ytvnr) [reply src/management/management.module.ts]: resolved by 5272088

**Review by phiz71 — APPROVED** [review]

File: src/management/api/apiAdmin.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (phiz71) [reply src/management/api/apiAdmin.controller.ts:132]: > if( '' ) evaluates to false

I had forgotten this  😅 .
Ok then.

**Review by phiz71 — APPROVED** [review]

**Review by phiz71 — APPROVED** [review]
